### ISSUE TITLE
Unable to run xiccd with latest glib2

### ISSUE THREAD

**Issue by sQVe** [agalakhov/xiccd#18]
_I'd to take the oppertunity and give you a huge thanks for this awesome daemon. A live saver for someone like me that wants to run profiles on a WM (i3) system._

I noted that my color profile wasn't loading properly and found that `xiccd` tried running but got a **segmentation fault**:
```
% xiccd
[1]    3534 segmentation fault (core dumped)  xiccd
%
```
This also printed the following in `dmesg`:
```
[  215.377072] traps: xiccd[3534] general protection ip:7effb10a431f sp:7ffdfc845c18 error:0 in libgobject-2.0.so.0.5400.3[7effb106d000+54000]
```
`glib2` owns the mentioned file:
```
% pacaur -Qo /usr/lib/libgobject-2.0.so.0.5400.3                                                                                 ~
/usr/lib/libgobject-2.0.so.0.5400.3 is owned by glib2 2.54.3+2+g94b38beff-1
```

______

I had to downgrade two versions of `glib2` (to version `glib2-2.54.2-2`) for not `xiccd` to crash. I would love some help in investigating this and if this is something that should be owned by `xiccd` or `glib2`.

**Comment by agalakhov** [comment]
Thank you. Does just recompiling xiccd help?

**Comment by sQVe** [comment]
@agalakhov I just tried recompiling and it sadly did no difference at all. Still getting segmentation fault with `glib2` version `2.54.3-1` and `2.54.3-2`.

**Comment by sQVe** [comment]
@agalakhov Any news on this issue? I have to hold back updating `glib2`, which I would prefer not to.

**Comment by agalakhov** [comment]
I just checked it. The crash happens inside libcolord.

**Comment by sQVe** [comment]
@agalakhov Okay. So this is something that I would raise with **colord** instead then?

**Comment by agalakhov** [comment]
Probably. It still may be my mistake, but the actual cause is only known to colord developers.

**Comment by hughsie** [comment]
Got a backtrace?

**Comment by agalakhov** [comment]
Not really, but it's g_object_unref() inside ??() from /usr/lib/libcolord.so.2

**Comment by hughsie** [comment]
That really doesn't help.

**Comment by guzzisti** [comment]
Hi, same issue for me after Manjaro updated glib2 to 2.54.3+2+g94b38beff-1.

**Comment by crisisinaptica** [comment]
Hello, 
Same issue on Arch Linux, glib2 version 2.54.3+2+g94b38beff-1

**Comment by Lompik** [comment]
attached backtrace with "colord 1.4.1+6+gf08c25d-1" and "glib2 2.54.3+2+g94b38beff-1", xiccd 0.2.4
[xiccd.log](https://github.com/agalakhov/xiccd/files/1784817/xiccd.log)

**Comment by es20490446e** [comment]
Same issue.

**Comment by sQVe** [comment]
@hughsie Did you see the above attached backtrace?

**Comment by hughsie** [comment]
@sQVe I did, but I have no time to work on the problem this week I'm afraid.

**Comment by sQVe** [comment]
Holding back `glib2` isn't a good _quick fix_ anymore as there are already other, important, packages that need the updated `glib2`.

@hughsie I understand! It would be awesome if you had the time to give this a look in the near future. I can no longer hold back `glib2` so my calibration setup is essentially fully broken at the moment.

**Comment by hughsie** [comment]
I think this is an xiccd bug. If you comment out the `g_object_unref (profile);` in `update_profile_cb` then there's no crash. xiccd doesn't actually ref the profile object anywhere. `cd_client_get_profiles_finish()` returns with container scope, although the gtk-doc helpfully is wrong too, and in `cd_profile_added_sig()` it's only signal emission, so no object ref either.

**Comment by agalakhov** [comment]
Thank you, I'll try to figure out.

**Comment by guzzisti** [comment]
works for me too with glib2 2.56.0

**Comment by es20490446e** [comment]
Although now it echoes various [critical messages](https://github.com/agalakhov/xiccd/issues/19), even when the daemon still works.

**Comment by sQVe** [comment]
@es20490446e As far as I know it did print those warnings even before the `glib2` update. So it shouldn't be related to this issue.
